bigframes/operations/semantics.py

```diff
@@ -50,8 +50,61 @@ def filter(self, instruction: str, model):
                 columns are referred to.
         """
 
+        output_instruction = "Based on the provided context, reply to the following claim by only True or False:"
+
+        from bigframes.dataframe import DataFrame
+
+        results = typing.cast(
+            DataFrame, model.predict(self._make_prompt(instruction, output_instruction))
+        )
+
+        return self._df[
+            results["ml_generate_text_llm_result"].str.lower().str.contains("true")
+        ]
+
+    def map(self, instruction: str, model):
+        """
+        Filters the DataFrame with the semantics of the user instruction.
+
+        Args:
+            instruction:
+                An instruction on how to filter the data. This value must contain
+                column references by name, which should be wrapped in a pair of braces.
+                For example, if you have a column "food", you can refer to this column
+                in the instructions like:
+                "The {food} is healthy."
+
+            model:
+                A GeminiTextGenerator provided by Bigframes ML package.
+
+        Returns:
+            DataFrame with attached mapping results.
+
+        Raises:
+            NotImplementedError: when the semantic operator experiment is off.
+            ValueError: when the instruction refers to a non-existing column, or when no
+                columns are referred to.
+        """
+        output_instruction = (
+            "Based on the provided contenxt, answer the following instruction:"
+        )
+
+        from bigframes.series import Series
+
+        results = typing.cast(
+            Series,
+            model.predict(self._make_prompt(instruction, output_instruction))[
+                "ml_generate_text_llm_result"
+            ],
+        )
+
+        from bigframes.core.reshape import concat
+
+        return concat([self._df, results.rename("map_result")], axis=1)
```

Comment:
I am not sure which one is better:
1. Find a special name, such as `_semantic_map_result_<index>`? Also check if `_semantic_map_result_<index>` is a new column for the existing df.
2. Ask user to provide a custom column name?

Reply:
Asking for column name inputs sounds better